collector/ksmd_linux.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !noksmd
+
+package collector
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	ksmdSubsystem = "ksmd"
+)
+
+var (
+	ksmdFiles = []string{"merge_across_nodes", "pages_shared", "pages_sharing", "pages_to_scan",
+		"pages_unshared", "pages_volatile", "run", "sleep_millisecs"}
+)
+
+type ksmdCollector struct {
+	fullScans    prometheus.Counter
```

Comment:
It's better to use `ConstMetrics` in the update rather than maintaining state across scrapes.